src/containers/register.js

```diff
@@ -1,15 +1,166 @@
 import React from "react";
 
-class RegisterContainer extends React.Component {
+import Button from "../components/Button";
+import Login from "../components/Login";
+import { Input, Row, Col } from "react-materialize";
+
+import { push } from "react-router-redux";
+import { connect } from "react-redux";
+import linkState from "linkstate";
+
+import { LoginUser, RegisterCar } from "../reducers/register.js";
+
+class Register extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      login: false,
+      register_car: true,
+      car: {
+        make: "",
+        model: "",
+        year: "",
+        color: ""
+      },
+      err: false
+    };
+
+    this.login = this.login.bind(this);
+    this.registerCar = this.registerCar.bind(this);
+  }
+
+  async login() {
+    const { err, data } = await this.props.login();
+    if (err) {
+      this.setState(prevState => {
+        return Object.assign({}, prevState, {
+          err
+        });
+      });
+      return;
+    }
+    this.setState(prevState => {
+      return Object.assign({}, prevState, {
+        err: false,
+        login: false,
+        register_car: true
+      });
+    });
   }
+
+  async registerCar() {
+    const { err, car } = await this.props.registerCar(this.state.car);
+    if (err) {
+      this.setState(prevState => {
+        return Object.assign({}, prevState, {
+          err
+        });
+      });
+      return;
+    }
+    await this.setState(prevState => {
+      return Object.assign({}, prevState, {
```

Comment:
I think this is not necessary as I think that setState is immutable by default

Reply:
i agree with this